client/src/main/java/io/pravega/client/auth/PravegaAuthHandler.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.auth;
+
+import java.util.Map;
+
+public interface PravegaAuthHandler {
+
+    enum  PravegaAccessControlEnum {
+        NONE,
+        READ,
+        READ_UPDATE
+    }
+
+    /**
+     * Returns name of the handler.
+     * @return The unique name assigned to the handler.
+     */
+    String getHandlerName();
+
+    /**
+     * Authenticates a given request.
+     *
+     * @param headers the key-value pairs passed through grpc.
+     * @return Returns true when the user is authenticated.
+     */
+    boolean authenticate(Map<String, String> headers);
+
+    /**
+     * Authorizes the access to a given resources.
+     * @param resource the resource that needs to be accessed.
+     * @param headers the context for authorization.
+     * @return The level of authorization. Throws exception if not authorized.
+     */
+    PravegaAccessControlEnum authorize(String resource, Map<String, String> headers);
+
+    /**
+     * Sets the configuration. The auth handler can extract its config from this.
+     * @param serverConfig The server configuration.
+     */
+    void setServerConfig(Object serverConfig);
```

Comment:
i think its better to only have a getServerConfig method on the interface.

the implementation can set the config however which way they deem fit.. interface should govern that there should be some property that can be retrieved via the getter method.. and that can used in the business logic agnostic to implementation of interface.. 
a setter on the interface, esp without a getter does not serve any purpose from users of the interface.

Reply:
This is to pass controller specific configurations to the implementation. The implementation can definitely set the config how ever they see fit.
This is in case the implementation needs access to controller properties.

Reply:
It is not obvious when this value will be set by the caller and available for use by the implementation.